products/gkehub/api.yaml

```diff
@@ -0,0 +1,119 @@
+    min_version: beta
+    name: 'Membership'
+    base_url: "projects/{{project}}/locations/global/memberships"
+    create_url: "projects/{{project}}/locations/global/memberships?membershipId={{membership_id}}"
+    update_url: "projects/{{project}}/locations/global/memberships/{{membership_id}}"
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Membership contains information about a member cluster.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/gkehub/docs/'
+      api: 'https://cloud.google.com/gkehub/docs/reference/rest/v1beta1/projects.locations.memberships'
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        kind: 'gkehub#operation'
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: 'true'
+        allowed:
+          - 'true'
+          - 'false'
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error/errors'
+        message: 'message'
+
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'membershipId'
+        description: |
+          The client-provided identifier of the membership.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: |
+          The unique identifier of the membership.
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          The name of this entity type to be displayed on the console.
+        required: true
+        input: true
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: |
+          Labels to apply to this membership.
+      - !ruby/object:Api::Type::String
+        name: 'externalId'
+        description: |
+          An externally-generated and managed ID for this Membership.
+          This ID may still be modified after creation but it is not recommended to do so.
+          The ID must match the regex: `[a-zA-Z0-9][a-zA-Z0-9_\-\.]*` If this Membership
+          represents a Kubernetes cluster, this value should be set to the UUID of the
+          kube-system namespace object.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'uniqueId'
+        description: |
+          Output only. Google-generated UUID for this resource.
+        output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'endpoint'
+        input: true
+        description: |
+          If this Membership is a Kubernetes API server hosted on GKE, this is a self link to its GCP resource.
```

Comment:
i) Does v1 API only support GKE on GCP?
https://cloud.google.com/gkehub/docs/reference/rest/v1beta1/projects.locations.memberships does not seem to exist.

Reply:
That's correct.  And yes, that's part of why we're waiting for the end of September, to give them time to publish the docs.